book/src/future/precompiles.md

```diff
@@ -0,0 +1,15 @@
+# Precompiles
+Precompiles are highly optimized SNARK gadgets which can be invoked from the high-level programming language of the VM user. These gadgets are much more efficient for the prover than compiling down to the underlying ISA by exploiting the structure of the workload. In practice zkVMs use these for heavy cryptographic operations such as hash functions, signatures and other elliptic curve arithmetic.
+
+By popular demand, Jolt will support these gadgets as well. The short term plan is to optimize for minimizing Jolt-core development resources rather than optimal prover speed.
+
+Precompile support plan:
+1. RV32 library for supported precompiles
+2. Tracer picks up syscalls, sets relevant flag bits and loads memory accordingly
+3. Individual (uniform) Spartan instance for each precompile, repeated over `trace_length` steps
```

Comment:
do we know if we're doing one Spartan instance per precompile? Or one for all the precompiles?

Reply:
We're currently thinking on per precompile, but flexible depending on Justin's review